src/NLog/Internal/FileAppenders/FileAppenderCache.cs

```diff
@@ -142,7 +142,7 @@ public string ArchiveFilePatternToWatch
                     if (!string.IsNullOrEmpty(_archiveFilePatternToWatch))
                     {
                         string directoryPath = Path.GetDirectoryName(_archiveFilePatternToWatch);
-                        if (string.IsNullOrEmpty(directoryPath))
+                        if (!string.IsNullOrEmpty(directoryPath))
                             _externalFileArchivingWatcher.StopWatching(directoryPath);
```

Comment:
Can you also change `StopWatching(directoryPath)` to `StopWatching(_archiveFilePatternToWatch)`, so it matches `_externalFileArchivingWatcher.Watch(_archiveFilePatternToWatch)`.